src/components/JaegerIntegration/JaegerResults/SpanDetail.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Card, CardBody, Grid, GridItem, Title } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
+import { Span } from '../../../types/JaegerInfo';
+import history from '../../../app/History';
+import { PfColors } from '../../Pf/PfColors';
+import { KialiAppState } from '../../../store/Store';
+import { connect } from 'react-redux';
+import { serverConfig } from '../../../config';
+import { SpanTable } from './SpanTable';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+
+interface SpanDetailProps {
+  spans: Span[];
+  namespaceSelector: boolean;
+}
+
+interface SpanDetailState {
+  spanSelected?: Span;
+  isModalOpen: boolean;
+}
+
+export class SpanDetailC extends React.Component<SpanDetailProps, SpanDetailState> {
+  constructor(props: SpanDetailProps) {
+    super(props);
+    this.state = { isModalOpen: false };
+  }
+
+  onClickSpan = (span?: Span) => {
+    this.setState(({ isModalOpen }) => ({
+      isModalOpen: !isModalOpen,
+      spanSelected: span
+    }));
+  };
+
+  goService = (service: string = this.props.spans[0].process.serviceName) => {
+    const ns = service.split('.')[1] || serverConfig.istioNamespace;
+    const srv = service.split('.')[0];
+    return '/namespaces/' + ns + '/services/' + srv;
+  };
+
+  render() {
+    console.log(this.props.spans);
```

Comment:
Probably this "console.log" entries can go away once the PR is debugged and in good shape.